src/functions/parse.js

```diff
@@ -0,0 +1,83 @@
+import and from './and'
+import empty from './empty'
+import equals from './equals'
+import every from './every'
+import greaterThanOrEquals from './greater-than-equals'
+import greaterThan from './greater-than'
+import lessThanOrEquals from './less-than-equals'
+import lessThan from './less-than'
+import none from './none'
+import not from './not'
+import or from './or'
+import regex from './regex'
+import some from './some'
+import typeOf from './type-of'
+import xor from './xor'
+
+export function buildRule (jsonRule) {
+  const predicate = Object.keys(jsonRule)[0]
+  switch (predicate) {
+    case 'empty':
```

Comment:
One suggestion in this case is to create a Literal Object.
For instance:
```
const rule = {
 "empty": empty(...jsonRule[predicate]),
 "equals": equals(...jsonRule[predicate]),
...
}

return rule[predicate]
```

Reply:
@rondymesquita I like that. I chose switch so we could handle the default case (which hopefully are mostly limited to custom predicates that this feature won't support, at least for now) explicitly, and allow for comments, but certainly am not committed to the design pattern. The object literal would probably end up being less lines of code. We'd need to add a little check to make sure the predicate exists. Could be something like 

```javascript
return rules[predicate] || `${predicate} not found`
```

Are there other advantages I'm not thinking of?

Reply:
I would say that the code functionality is the same. With literal object you just have less code repetitions (case keyword), and also avoid multiples returns in same function (return in each 'case'). When a function grows and you have multiples returns on it, it might become complex to read and to track where output is coming from. It is just a suggestion.

Reply:
And, I think yes: this `return rules[predicate] || '${predicate} not found'` should work fine. 😄

Reply:
@ rondymesquita on closer inspection this won't work as cleanly as this because when we declare the `rules` object it needs to build each rule, not just the one we need. so in this example

```javascript
const rules = {
  equals: equals(...jsonRule[predicate]),
  and: and(...jsonRule[predicate].map(x => buildRule(x))),
}

return rules['equals']
```

we evaluate the `equals` rule, and the `and` rule. This is an issue because the predicate they need is a different shape. We would need to write the `and` line to safely check to make sure predicate is an array before evaluating. But that also requires more compute cycles for no reason.

I believe I'm going to leave the switch syntax so we only prepare the predicate we need for each rule, not all the predicates for each rule.

Does that make sense?

If you disagree feel free to submit a PR, I'm happy to take a look.

Reply:
oh! I see your point! yes, I agree with you.